src/components/Header/components/NavButtonsRightCorner/navButtons.js

```diff
@@ -0,0 +1,324 @@
+import React from 'react';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemText from '@material-ui/core/ListItemText';
+import { NavLink } from 'react-router-dom';
+import user from '../../../../services/user';
+
+/**
+ * Creates the Links button.
+ *
+ * Depending on the status of the network, authentication, and user role, the Links button is created.
+ *
+ * @param {String} networkStatus The status of the network. Either 'online' or 'offline'.
+ * @param {Function} onLinkClickOne Determines what occurs when the user clicks on this button.
+ * @param {Function} onLinkClickTwo Determines what occurs when the user clicks on this button
+ *                                  after onLinkClickOne occurs.
+ * @param {Function} openDialogBox Sets the data of a dialog box to be shown if the user clicks on
+ *                                this button.
+ *
+ * @return {JSX} The JSX of the Links button.
+ */
+export function createLinksButton(networkStatus, onLinkClickOne, onLinkClickTwo, openDialogBox) {
+  let linksButton =
+    // Network Status: Online
+    networkStatus === 'online' ? (
+      <ListItem
+        divider
+        button
+        onClick={() => {
+          onLinkClickOne();
+          onLinkClickTwo();
+        }}
+      >
+        <ListItemText primary="Links" />
+      </ListItem>
+    ) : (
+      // Network Status: Offline
+      <div
+        onClick={() => {
+          openDialogBox('offline', '');
+        }}
+      >
+        <ListItem divider button disabled={true}>
+          <ListItemText primary="Links" />
+        </ListItem>
+      </div>
+    );
+
+  return linksButton;
+}
+
+/**
+ * Creates the Feedback button.
+ *
+ * Depending on the status of the network, the Feedback button is created.
+ *
+ * @param {String} networkStatus The status of the network. Either 'online' or 'offline'.
+ * @param {Function} onLinkClick Determines what occurs when the user clicks on this button.
+ * @param {Function} openDialogBox Sets the data of a dialog box to be shown if the user clicks on
+ *                                this button.
+ *
+ * @return {JSX} The JSX of the Feedback button.
+ */
+export function createFeedbackButton(networkStatus, onLinkClick, openDialogBox) {
+  let feedbackButton =
+    // Network Status: Online
+    networkStatus === 'online' ? (
```

Comment:
I know what you mean lol. I wanted to have a global variable that would take care of the network and authentication. It is possible to have a global variable for the network but it's not possible for authentication since these functions are not apart of a component. In other words, I can't pass authentication as a prop. I can get the network status from local storage and set it as a global variable, but I found it easier at the time to just pass it in from the Header component as a parameter along with authentication.